src/dotnet/APIView/APIView/Model/TokenTreeModel.cs

```diff
@@ -0,0 +1,250 @@
+        {
+            var token = new StructuredToken();
+            token.Id = id;
+            token.Kind = StructuredTokenKind.Content;
+            return token;
+        }
+
+        public static StructuredToken CreateSpaceToken()
+        {
+            var token = new StructuredToken();
+            token.Kind = StructuredTokenKind.NoneBreakingSpace;
+            return token;
+        }
+
+        public static StructuredToken CreateParameterSeparatorToken()
+        {
+            var token = new StructuredToken();
+            token.Kind = StructuredTokenKind.ParameterSeparator;
+            return token;
+        }
+
+        public static StructuredToken CreateTextToken(string value, string id = null)
+        {
+            var token = new StructuredToken(value);
+            if (!string.IsNullOrEmpty(id))
+            {
+                token.Id = id;
+            }
+            token.RenderClasses.Add("text");
+            return token;
+        }
+
+        public static StructuredToken CreateKeywordToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("keyword");
+            return token;
+        }
+
+        public static StructuredToken CreateKeywordToken(SyntaxKind syntaxKind)
+        {
+            return CreateKeywordToken(SyntaxFacts.GetText(syntaxKind));
+        }
+
+        public static StructuredToken CreateKeywordToken(Accessibility accessibility)
+        {
+            return CreateKeywordToken(SyntaxFacts.GetText(accessibility));
+        }
+
+        public static StructuredToken CreatePunctuationToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("punc");
+            return token;
+        }
+
+        public static StructuredToken CreatePunctuationToken(SyntaxKind syntaxKind)
+        {
+            return CreatePunctuationToken(SyntaxFacts.GetText(syntaxKind));
+        }
+
+        public static StructuredToken CreateTypeNameToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("tname");
+            return token;
+        }
+
+        public static StructuredToken CreateMemberNameToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("mname");
+            return token;
+        }
+
+        public static StructuredToken CreateLiteralToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("literal");
+            return token;
+        }
+
+        public static StructuredToken CreateStringLiteralToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("sliteral");
+            return token;
+        }
+    }
+
+    public class APITreeNode
+    {
+        [JsonPropertyName("t")]
```

Comment:
If you are planning to serialize using binary serialization then let's make these properties name a bit more clear than aggressive shortening.